src/pages/lessons/[slug].tsx

```diff
@@ -221,13 +222,40 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
     )
   }, [currentPlayerState])
 
-  const checkAutoPlay = () => {
+  const checkAutoPlay = async () => {
     if (autoplay && nextLesson) {
       console.debug('autoplaying next lesson', {nextLesson})
       track('autoplaying next video', {
         video: nextLesson.slug,
       })
-      router.push(nextLesson.path)
+
+      if (isFullscreen) {
+        const loadedLesson = await loadLesson(
+          nextLesson.slug,
+          getAccessTokenFromCookie(),
+        )
+
+        console.debug('full screen authed video loaded', {video: loadedLesson})
+
+        const mediaUrls = {
+          hls_url: loadedLesson.hls_url,
+          dash_url: loadedLesson.dash_url,
+        }
+
+        send({
+          type: 'VIEW',
+          lesson: loadedLesson,
+        })
+
+        setMedia(mediaUrls)
+        setLesson(loadedLesson)
+
+        router.push(loadedLesson.path, undefined, {
```

Comment:
this causes a flash of the whole UI, which isn't nice, but also it keeps playing so that's cool